### PR TITLE
Bug 1377168 - XCUICompactModeUITests

### PR DESCRIPTION
**Pull request by ugarlapati**

### CODE REVIEW

**Comment by farhanpatel** [comment]
### SwiftLint found issues

#### Warnings

File | Line | Reason |
| --- | ----- | ----- |
CompactModeUITests.swift | 56 | Limit vertical whitespace to a single empty line. Currently 2. 


<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

File: XCUITests/CompactModeUITests.swift [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        XCTAssertTrue(app.collectionViews.cells.countForHittables <= 4)
+        
+        compactModeOn()
+
+        //CollectionView visible cells count should be 6
+        XCTAssertTrue(app.collectionViews.cells.countForHittables == 6)
+    }
+}
+
+extension XCUIElementQuery {
```
Comment (npark-mozilla) [inline XCUITests/CompactModeUITests.swift]: @ugarlapati are we using this extension?

File: XCUITests/CompactModeUITests.swift [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+            "www.yahoo.com"
+        ]
+
+        //Open the array of urls in 6 different tabs
+        for i in 0..<urls.count {
+            if i > 0 {
+                openNewTab()
+            }
+            enterAndOpenUrl(url: urls[i])
+        }
```
Comment (npark-mozilla) [inline XCUITests/CompactModeUITests.swift]: let's simplify this, like

enterAndOpenUrl(url: urls[i])
for i in 1..<urls.count {
               openNewTab()
            enterAndOpenUrl(url: urls[i])
        }

Easier to understand this way

File: XCUITests/CompactModeUITests.swift [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        navigator = nil
+        app = nil
+        super.tearDown()
+    }
+    
+    //Set the compact mode to OFF in settings screen
+    private func compactModeOff() {
+        let app = XCUIApplication()
+        app.buttons["TabTrayController.menuButton"].tap()
+        app.collectionViews.cells["SettingsMenuItem"].tap()
```
Comment (npark-mozilla) [inline XCUITests/CompactModeUITests.swift]: I think we should use Screengraph here, like navigator.goto(SettingsScreen)

File: XCUITests/CompactModeUITests.swift [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+        app.tables["AppSettingsTableViewController.tableView"].switches["Use Compact Tabs"].tap()
+        app.navigationBars["Settings"].buttons["AppSettingsTableViewController.navigationItem.leftBarButtonItem"].tap()
```
Comment (npark-mozilla) [inline XCUITests/CompactModeUITests.swift]: also use screengraph here, like navigator.goto()

File: XCUITests/CompactModeUITests.swift [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+    }
+    
+    //Set the compact mode to ON in settings screen
+    private func compactModeOn() {
+        app.buttons["TabTrayController.menuButton"].tap()
+        app.collectionViews.cells["SettingsMenuItem"].tap()
```
Comment (npark-mozilla) [inline XCUITests/CompactModeUITests.swift]: Screengraph should be used here as well

**Comment by ugarlapati** [comment]
Yeah, using this extension to identify the visible cells count.

File: Client.xcodeproj/project.pbxproj [hunk 6]
```diff
@@ -4384,7 +4387,7 @@
 					};
 					D39FA15E1A83E0EC00EE869C = {
 						CreatedOnToolsVersion = 6.1.1;
-						DevelopmentTeam = 43AQ936H96;
+						DevelopmentTeam = 9G8J6YA743;
```
Comment (npark-mozilla) [inline Client.xcodeproj/project.pbxproj]: this should be reverted as well

File: XCUITests/CompactModeUITests.swift (hunk 1 above)
Comment (npark-mozilla) [inline XCUITests/CompactModeUITests.swift]: I made changes as below, and it seems to run reliabily.

/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at http://mozilla.org/MPL/2.0/. */

import XCTest

class CompactModeUITests: BaseTestCase {
        
    var navigator: Navigator!
    var app: XCUIApplication!
    
    override func setUp() {
        super.setUp()
        app = XCUIApplication()
        navigator = createScreenGraph(app).navigator(self)
    }
    
    override func tearDown() {
        navigator = nil
        app = nil
        super.tearDown()
    }
    
    //Set the compact mode to OFF in settings screen
    private func compactModeOff() {
        //let app = XCUIApplication()
        // app.buttons["TabTrayController.menuButton"].tap()
        // app.collectionViews.cells["SettingsMenuItem"].tap()
        navigator.goto(SettingsScreen)
        app.tables["AppSettingsTableViewController.tableView"]
            .switches["Use Compact Tabs"].tap()
        navigator.goto(TabTray)
        //app.navigationBars["Settings"].buttons["AppSettingsTableViewController.navigationItem.leftBarButtonItem"].tap()
    }
    
    //Set the compact mode to ON in settings screen
    private func compactModeOn() {
        //app.buttons["TabTrayController.menuButton"].tap()
        //app.collectionViews.cells["SettingsMenuItem"].tap()
        navigator.goto(SettingsScreen)
        app.tables["AppSettingsTableViewController.tableView"].switches["Use Compact Tabs"].tap()
        navigator.goto(TabTray)
        //app.navigationBars["Settings"].buttons["AppSettingsTableViewController.navigationItem.leftBarButtonItem"].tap()
    }
    
    /*
    //Opens a new tab
    private func openNewTab() {
        app.buttons["URLBarView.tabsButton"].tap()
        app.buttons["TabTrayController.addTabButton"].tap()
        app.collectionViews.cells["TopSitesCell"].collectionViews.containing(.cell, identifier:"TopSite").element.tap()
    }
    */
    /*
    //Enters the url in the browser tab
    private func enterAndOpenUrl(url: String) {
        app.textFields["url"].tap()
        waitforExistence(app.textFields["address"])
        app.textFields["address"].typeText(url)
        app.buttons["Go"].tap()
    }
    */
    func testCompactModeUI() {
        //Dsimiss intro screen
        dismissFirstRunUI()

        //Creating array of 6 urls
        let urls: [String] = [
            "www.google.com",
            "www.facebook.com",
            "www.youtube.com",
            "www.amazon.com",
            "www.twitter.com",
            "www.yahoo.com"
        ]

        //Open the array of urls in 6 different tabs
        //enterAndOpenUrl(url: urls[0])
        loadWebPage(urls[0])
        for i in 1..<urls.count {
            //openNewTab()
            navigator.createNewTab()
            loadWebPage(urls[i])
        }

        //Navigate to tabs tray
        navigator.goto(TabTray)
      
        //Wait until the cells show up
        //CollectionView visible cells count should be 6
        let exists = NSPredicate(format: "countForHittables > 1")
        expectation(for: exists, evaluatedWith: app.collectionViews.cells, handler: nil)
        waitForExpectations(timeout: 10, handler: nil)
        XCTAssertTrue(app.collectionViews.cells.countForHittables == 6)

        compactModeOff()

        //CollectionView visible cells count should be less than or equal to 4
        XCTAssertTrue(app.collectionViews.cells.countForHittables <= 4)
        
        compactModeOn()

        //CollectionView visible cells count should be 6
        XCTAssertTrue(app.collectionViews.cells.countForHittables == 6)
    }
}

extension XCUIElementQuery {
    var countForHittables: UInt {
        return UInt(allElementsBoundByIndex.filter { $0.isHittable }.count)
    }
}

File: Client.xcodeproj/project.pbxproj [hunk 7]
```diff
@@ -6207,7 +6211,7 @@
 			buildSettings = {
 				CODE_SIGN_ENTITLEMENTS = "$(SRCROOT)/Extensions/Entitlements/FennecEnterprise.entitlements";
 				"CODE_SIGN_IDENTITY[sdk=iphoneos*]" = "iPhone Developer";
-				DEVELOPMENT_TEAM = 9G8J6YA743;
+				DEVELOPMENT_TEAM = 43AQ936H96;
```
Comment (npark-mozilla) [inline Client.xcodeproj/project.pbxproj]: these IDs should be reverted

File: Client.xcodeproj/project.pbxproj [hunk 8]
```diff
@@ -7268,7 +7272,7 @@
 			buildSettings = {
 				ALWAYS_SEARCH_USER_PATHS = NO;
 				APPLICATION_EXTENSION_API_ONLY = NO;
-				APP_IDENTIFIER_PREFIX = 9G8J6YA743;
+				APP_IDENTIFIER_PREFIX = 43AQ936H96;
```
Comment (npark-mozilla) [inline Client.xcodeproj/project.pbxproj]: these IDs should be reverted

File: Client.xcodeproj/project.pbxproj [hunk 9]
```diff
@@ -7752,7 +7756,7 @@
 				CURRENT_PROJECT_VERSION = 1;
 				DEBUG_ACTIVITY_MODE = "";
 				"DEBUG_ACTIVITY_MODE[sdk=iphonesimulator*]" = disable;
-				DEVELOPMENT_TEAM = 9G8J6YA743;
+				DEVELOPMENT_TEAM = 43AQ936H96;
```
Comment (npark-mozilla) [inline Client.xcodeproj/project.pbxproj]: these IDs should be reverted

**Pull request merged**